superset/views/api.py

```diff
@@ -0,0 +1,32 @@
+# pylint: disable=R
+import json
+
+from flask import g, request
+from flask_appbuilder import expose
+from flask_appbuilder.security.decorators import has_access_api
+
+from superset import appbuilder, security_manager
+from superset.common.query_context import QueryContext
+from superset.models.core import Log
+from .base import api, BaseSupersetView, data_payload_response, handle_superset_exception
+
+
+class Api(BaseSupersetView):
+    @Log.log_this
+    @api
+    @handle_superset_exception
+    @has_access_api
+    @expose('/v1/query/', methods=['POST'])
+    def query(self):
+        """
+        Takes a query_obj constructed in the client and returns payload data response
+        for the given query_obj.
+        """
+        query_context = QueryContext(**json.loads(request.form.get('query_context')))
+        print(query_context)
+        security_manager.assert_datasource_permission(query_context.datasource, g.user)
+        payload_json = query_context.query_object.get_data()
```

Comment:
Since now `query_object` won't have the datasource info, we may need to move `get_data` as a function in `query_context`

Reply:
but I think I can revise it later in my PR.

Reply:
👍

Reply:
Actually, it is trivial to move the empty implementation, so I moved the method into `QueryContext`. 😄